spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -240,8 +244,30 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
     }
   }
 
+  public boolean isHiveCatalogConfigValid(CaseInsensitiveStringMap options, StringBuilder errorMsg) {
+    String hadoopConfUri = SparkContext.getOrCreate().conf().get("spark.hadoop.hive.metastore.uris");
+    String catalogConfUri = options.get("uri");
+    if (catalogConfUri != null) {
+      LOG.warn("Don't set uri for SparkSessionCatalog" +
+              "set it only in hive conf");
+      if (!hadoopConfUri.equalsIgnoreCase(catalogConfUri)) {
+        errorMsg.append("Cannot set uri for SparkSessionCatalog: " +
+                "conflicts with Hive conf URI");
+        return false;
+      }
+    }
+    return true;
+  }
+
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
+    if (options.containsKey("type") && options.get("type").equalsIgnoreCase("hive")) {
+      StringBuilder errorMsg = new StringBuilder();
+      if (!isHiveCatalogConfigValid(options, errorMsg)) {
+        throw new UnsupportedOperationException(errorMsg.toString());
+      }
+
```

Comment:
Style: there should be no newline before the block ends, but there should be a newline between a control flow block and the next statement.

Reply:
changed